tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -7138,6 +7189,28 @@ bool SpirvEmitter::isBufferTextureIndexing(const CXXOperatorCallExpr *indexExpr,
   return false;
 }
 
+bool SpirvEmitter::isDescriptorHeap(const CXXOperatorCallExpr *indexExpr,
```

Comment:
I prefer it when a function does 1 thing. I know the spir-v backend has lots of function like this, but could be split this into `isDescritorHeap`, which return true of false, as the name suggests. Then add another function to get the base and index.

I find it hard to read code when a function with a name like this has other return values.